File: zeroJudge/d809.py
import sys


def split(p):
        p1, p2 = p.split()
        return int(p1), int(p2)

def triangleArea(p1, p2, p3):
        a , b = split(p1)
        c , d = split(p2)
        e , f = split(p3)
        return abs(a * d + c * f + b * e - b * c - d * e - a * f)

while True:
        line = sys.stdin.readline().strip()
        if line == "":
                break
        num = int(line)
        points = []
        for i in range(num):
                point = sys.stdin.readline().strip()
                points.append(point)

        maxArea = 0
        for i in range(0, num):
                for j in range(i + 1, num):
                        for k in range(j + 1, num):
                                area = triangleArea(points[i], points[j], points[k])
                                if area > maxArea:                                        
                                        maxArea = area
        
        print "%.2f"%(float(maxArea/2.0))
                                                

File: zeroJudge/d809.c
#include <stdio.h>
#include <stdlib.h>



struct point
{
        int p1;
        int p2;
};


int triangleArea(struct point p1, struct point p2, struct point p3)
{
        int a = p1.p1;
        int b = p1.p2;
        int c = p2.p1;
        int d = p2.p2;
        int e = p3.p1;
        int f = p3.p2;
        return abs(a * d + c * f + b * e - b * c - d * e - a * f);
}
int main()
{
        struct point points[200];
        int n;
        while(scanf("%d", &n) != EOF)
        {
                int a,b;
                int i,j,k;
                for(i = 0 ; i < n ; i++)
                {
                        scanf("%d %d", &a, &b);
                        struct point temp;
                        temp.p1 = a;
                        temp.p2 = b;
                        points[i] = temp;
                }
                int maxValue = 0;
                for (i = 0 ; i < n ; i++)
                {
                        for(j = i + 1 ; j < n ; j++)
                        {
                                for(k = j + 1 ; k < n ; k++)
                                {
                                        int area = triangleArea(points[i], points[j], points[k]);
                                        if (area > maxValue)
                                        {
                                                maxValue = area;
                                        }
                                }
                        }
                }
                printf("%.2f\n", maxValue / 2.0);
        }
                      
        return 0;
}
